src/plugins/stake-plugins/currency/tronStakePlugin.ts

```diff
@@ -249,7 +276,7 @@ export const makeTronStakePlugin = async (pluginId: string): Promise<StakePlugin
         }
       }
 
-      const allocations: QuoteAllocation[] = [
+      allocations.push(
```

Comment:
Taking advantage of there being a block for unstake specific functionality

Reply:
I like to think of each "paragraph" of code to be responsible for a certain topic. If you look at the switch statement, it is the declaration of the `edgeTransaction` above. Just because that declaration has the same conditions for another declaration, does necessarily make a for a good argument to inserting more topics into this paragraph of code.

I get why this may have "felt right", which is why I personally dislike mutable variables declared above it's declaration with some large control-flow code. This sort of pattern is something I see often which is a style I had to get used to, but my gripe with it remains. Here's some examples:

```js
let sum = 0
if (someCondition) {
  for (const i = 0; i < items.length; ++i) {
      sum += items[i]
  }
} else if (someOtherCondition) {
  sum = somethingElse
}
```

This code is written to declare the value of `sum`, but because of the imperative approach used, it may not be obvious and so another developer may come along and think it's okay to insert some other logic in the conditions to "re-use the control flow":


```js
let sum = 0
if (someCondition) {
  for (const i = 0; i < items.length; ++i) {
      sum += items[i]
  }
  
  const bool = executeSomeFunction()
  if (bool) {
    unrelatedValue = sum/2
  }
} else if (someOtherCondition) {
  sum = somethingElse
}
```

From a readability perspective, to know what `sum` is declared as, we have to skip over some unrelated declaration inserted into `sum`'s declaration. This sort of code decay is _exactly_ what I fear whenever I see or write code in an imperative way.

But let's ask ourselves if the declaration for sum was using a declarative approach, would we make the same judgement?

```js
const sum = someCondition
  ? items.reduce((sum, items) => sum + items[i], 0)
  : somethingElse
```

Now because this declaration for `sum` is rigid, it's harder to insert some other code which confuses it's declaration or makes it harder to reason about. Would we ever think to somehow insert the definition of `unrelatedValue` into this declaration somehow? Not likely.

I'm tempted to rewrite the switch statement into a declaration for `edgeTransaction`  at this point.